src/proto3Main.ts

```diff
@@ -79,20 +79,21 @@ export function activate(ctx: vscode.ExtensionContext): void {
 
     vscode.languages.registerDocumentFormattingEditProvider('proto3', {
         provideDocumentFormattingEdits(document: vscode.TextDocument): vscode.TextEdit[] {
-
+            let args = [];
             let opts = { input: document.getText() };
 
             // In order for clang-format to find the correct formatting file we need to have cwd set appropriately
-            switch(document.uri.scheme) {
+            switch (document.uri.scheme) {
                 case "untitled": // File has not yet been saved to disk use workspace path
                     opts['cwd'] = vscode.workspace.rootPath;
+                    args.push(`--assume-filename=untitled.proto`);
```

Comment:
Typically, I only run formatting on save, so I'm not sure when this would happen. If there isn't a filename, providing any filename with a proto file extension seems to work.